src/main/java/org/reactome/release/Resource.java

```diff
@@ -0,0 +1,166 @@
+package org.reactome.release;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import com.google.gson.JsonParser;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.csv.CSVRecord;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Resource {
+	private final Logger logger = LogManager.getLogger();
+	private JsonObject resourceAsJson;
+
+	public Resource(CSVRecord resourceInfoCSV) {
+		this.resourceAsJson = convertCSVToJson(resourceInfoCSV);
+		checkMandatoryAttributesExist(this.resourceAsJson.toString());
+	}
+
+	public Resource(JsonObject resourceInfoJson) {
+		this.resourceAsJson = resourceInfoJson;
+		checkMandatoryAttributesExist(this.resourceAsJson.toString());
+	}
+
+	private JsonObject convertCSVToJson(CSVRecord resourceInfoCSV) {
+		String resourceInfoJson = convertCSVToJsonString(resourceInfoCSV);
```

Comment:
Why is it necessary to convert the CSV to JSON? Why can't the CSV be used directly?

Reply:
I wanted to make sure the Resource object has only one underlying data structure instead of two.  JSON was easier to work with.  I started with CSV initially, so the project supports both right now, but may be phasing out CSV.

Reply:
Ok.